src/Commands/Concerns/InstallsFrankenPhpDependencies.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Laravel\Octane\Commands\Concerns;
+
+use GuzzleHttp\Client;
+use Illuminate\Support\Facades\Http;
+use Laravel\Octane\FrankenPhp\Concerns\FindsFrankenPhpBinary;
+
+trait InstallsFrankenPhpDependencies
+{
+    use FindsFrankenPhpBinary;
+
+    /**
+     * Ensure the FrankenPHP binary is installed into the project.
+     */
+    protected function ensureFrankenPhpBinaryIsInstalled(): string
+    {
+        if (! is_null($frankenphpBinary = $this->findFrankenPhpBinary())) {
+            return $frankenphpBinary;
+        }
+
+        if ($this->confirm('Unable to locate FrankenPHP binary. Should Octane download the binary for your operating system?', true)) {
+            $this->downloadFrankenPhpBinary();
+
+            copy(__DIR__.'/../stubs/Caddyfile', base_path('Caddyfile'));
+        }
+
+        return base_path('frankenphp');
+    }
+
+    /**
+     * Download the latest version of the FrankenPHP binary.
+     *
+     * @return bool
+     */
+    protected function downloadFrankenPhpBinary()
+    {
+        $arch = php_uname('m');
+
+        $assetName = null;
+        if (PHP_OS_FAMILY === 'Linux') {
+            if ($arch !== 'x86_64') {
+                $assetName = 'frankenphp-linux-x86_64';
+            }
```

Comment:
This needs a second look.

And I would suggest replacing the whole section with a `match` clause for readability:
```php
$assetName = match (true) {
    PHP_OS_FAMILY === 'Linux' && $arch === 'x86_64' => 'frankenphp-linux-x86_64',
    PHP_OS_FAMILY === 'Darwin' => "frankenphp-mac-$arch",
    default => null,
};
```

Reply:
Good idea!